src/components/Auth/SignUp.tsx

```diff
@@ -1,14 +1,21 @@
 import React from "react";
 import { View } from "react-native";
 import styled from "styled-components";
-import Input from "../CommonUIElements/Input";
 import Button from "../CommonUIElements/Button";
+import ValidatedInput from "./ValidatedInput";
 import { Header, Title, Annotation, Body } from "./styled";
 import { signUp } from "../../actions/auth";
 import { connect } from "react-redux";
+import { validate } from "../../helpers/validator";
 
 interface IProps {
```

Comment:
can we do this
interface SignUpData { username: string; email: string; password: string }
interface IProps {
   signUp: (data: SignUpData) => void;
...
 error: SignUpData;
...